kiwi/config/functions.sh

```diff
@@ -1011,7 +1013,14 @@ function suseSetupProduct {
         elif [ -f "SUSE_${prod}.prod" ];then
             ln -sf "SUSE_${prod}.prod" baseproduct
         else
-            prod=$(find . -maxdepth 1 -name "*.prod" 2>/dev/null | tail -n 1)
+            find_prod() {
+                for f in $(ls .); do
+                    if [[ $f =~ \.prod$ ]]; then
+                        echo $f
+                    fi
+                done
+            }
+            prod=$(find_prod | tail -n 1)
```

Comment:
Can you explain why find should not be used ?

The tools used in the former implementation are provided by:

* findutils
* coreutils

After the change the tools needed are provided by

* coreutils(ls)
* python

Can you share your thoughts why we should change the code ?

Thanks

Reply:
`find` introduces another dependency that needs to be included in the buildroot for very minimal image builds, whereas using python, which is already in the buildroot, does not.

Reply:
I did not understand that part. You can build an image without python installed. As soon as scripts requires python you cannot have an image without python installed or do not use our script methods. I can't tell if find is more likely to exist on Linux than python, but one is core linux and the other is a programming language

Reply:
Yes, that's an oversight on my side. I definitely do **not** want the image to require python being installed.